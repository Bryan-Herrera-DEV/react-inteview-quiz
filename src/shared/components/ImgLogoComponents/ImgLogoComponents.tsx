import { ICommonComponent } from '@/shared/interfaces/CommonComponent.interface';
import { TLogos } from '../../interfaces/logos.interface';

import LogoCompleteBlueWhite from '../../assets/logos/ReactInterviewQuiz-blue-white.webp';
import LogoCompleteWhiteBlue from '../../assets/logos/ReactInterviewQuiz-white-blue.webp';
import Short_LogoCompleteBlueWhite from '../../assets/logos/Short_ReactInterviewQuiz-blue-white.webp';
import Short_LogoCompleteWhiteBlue from '../../assets/logos/Short_ReactInterviewQuiz-white-blue.webp';


// convert in required optional parameters of the IImgLogoComponents interface
interface IImgLogoComponents extends Required<Pick<ICommonComponent, 'className'>> {
    Logo: TLogos
}


const ImgLogoComponents = ({ Logo, className }: IImgLogoComponents) => {
    switch (Logo) {
        case 'LogoCompleteBlueWhite':
            return <img src={LogoCompleteBlueWhite} alt="LogoCompleteBlueWhite" className={className} />
        case 'LogoCompleteWhiteBlue':
            return <img src={LogoCompleteWhiteBlue} alt="LogoCompleteWhiteBlue" className={className} />
        case 'Short_LogoCompleteBlueWhite':
            return <img src={Short_LogoCompleteBlueWhite} alt="Short_LogoCompleteBlueWhite" className={className} />
        case 'Short_LogoCompleteWhiteBlue':
            return <img src={Short_LogoCompleteWhiteBlue} alt="Short_LogoCompleteWhiteBlue" className={className} />
        default:
            return <img src={LogoCompleteBlueWhite} alt="LogoCompleteBlueWhite" className={className} />
    }
}

export default ImgLogoComponents
