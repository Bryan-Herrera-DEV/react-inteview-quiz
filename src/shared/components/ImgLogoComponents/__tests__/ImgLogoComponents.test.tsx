import ImgLogoComponents from "../ImgLogoComponents";
import { render } from '@testing-library/react';

/**
 * @jest-environment jsdom
 */

describe('ImgLogoComponents', () => {
    it('should render', function() {
        const { container } = render(<ImgLogoComponents Logo="LogoCompleteBlueWhite" className="test" />);
        expect(container).toMatchSnapshot();
    });
});