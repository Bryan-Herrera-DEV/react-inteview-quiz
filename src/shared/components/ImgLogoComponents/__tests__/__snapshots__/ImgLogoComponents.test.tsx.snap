// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgLogoComponents should render 1`] = `
<div>
  <img
    alt="LogoCompleteBlueWhite"
    class="test"
  />
</div>
`;
